### ISSUE TITLE
Regression methods with parameters using Micronaut Nullability annotations are treated as executable methods in Micronaut Framework 3.6.4 and above

### ISSUE THREAD

**Issue by iiq374** [micronaut-projects/micronaut-core#8455]
### Expected Behavior

@NotNull annotation can be used on private methods

### Actual Behaviour

This regression seems to occur between Micronaut 3.5.5 and 3.6.3 - 
https://github.com/micronaut-projects/micronaut-core/issues/8164

```    @NotNull
    @SneakyThrows
    private BusinessTransactionStatus updateBusinessTxStatus(UUID paymentTxId, OffsetDateTime processAt,
                                                             BusinessTransactionStatus.Status status) {
...
}```

Gives compilation error:
> error: Method annotated with constraints but is declared private. Change the method to be non-private in order for AOP advice to be applied.

### Steps To Reproduce

Use @NotNull annotation on private method using Micronaut 3.5.5
Compile (success)
Update to Micronaut 3.6.4 or 3.7.3
Compile (failure / replication of error)

### Environment Information

_No response_

### Example Application

_No response_

### Version

3.6.4

**Comment by iiq374** [comment]
@sdelamo Would your patch for this against 4.0.x be portable back to the 3.6 and/or 3.7 branches ?

**Comment by graemerocher** [comment]
`@NotNull` is a validation constraint. The correct nullability annotations is `@NonNull`

**Comment by iiq374** [comment]
@graemerocher Thanks will attempt that change; was that a conscious change between 3.5.3 --> 3.6.0?

**Comment by graemerocher** [comment]
support for a bean being validated without declaring `@Validated` has been around for some time https://github.com/micronaut-projects/micronaut-core/commit/0f25fdd7b15fb34abadc08495e7b51d7375f90dd

Perhaps there was a regression somewhere where it wasn't the case in 3.5.3 but a bug was fixed in 3.6
